ingestion/src/test/java/feast/ingestion/ImportJobTest.java

```diff
@@ -180,91 +185,332 @@ public void runPipeline_ShouldWriteToRedisCorrectlyGivenValidSpecAndFeatureRow()
     options.setStoreJson(Collections.singletonList(JsonFormat.printer().print(redis)));
     options.setProject("");
     options.setBlockOnRun(false);
+  }
+
+  @BeforeClass
+  public static void setup() throws IOException, InterruptedException {
+    LocalKafka.start(
+        KAFKA_HOST,
+        KAFKA_PORT,
+        KAFKA_REPLICATION_FACTOR,
+        true,
+        ZOOKEEPER_HOST,
+        ZOOKEEPER_PORT,
+        ZOOKEEPER_DATA_DIR);
+    LocalRedis.start(REDIS_PORT);
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    LocalRedis.stop();
+    LocalKafka.stop();
+  }
+  // Ingestion test where feature rows match registered spec exactly.
+  @Test
+  public void runPipeline_ShouldWriteToRedisCorrectlyGivenValidSpecAndFeatureRow()
+      throws IOException, InterruptedException {
 
     List<FeatureRow> input = new ArrayList<>();
-    Map<RedisKey, FeatureRow> expected = new HashMap<>();
 
     LOGGER.info("Generating test data ...");
     IntStream.range(0, IMPORT_JOB_SAMPLE_FEATURE_ROW_SIZE)
         .forEach(
             i -> {
-              FeatureRow randomRow = TestUtil.createRandomFeatureRow(featureSet.getSpec());
-              RedisKey redisKey = TestUtil.createRedisKey(featureSet.getSpec(), randomRow);
+              FeatureRow randomRow = TestUtil.createFeatureRowWithDefaults(spec, defaultValues);
```

Comment:
Since this created FeatureRow is not random (the value is `default`) maybe the variable name should be updated.

Reply:
Also for other similar variables named `randomRow`